packages/x-date-pickers/src/internals/hooks/useField/useField.interfaces.ts

```diff
@@ -67,6 +84,8 @@ export interface FieldSection {
   query: string | null;
 }
 
+export type FieldSelectionSectionIndexes = { start: number; end: number } | null;
```

Comment:
> Would it be possible to have shortcuts and then don't need to provide imperative access to the sections with

We would not support format with several time the same section name in the format (which is very uncommon of course).
That's why everything is based on the index rather than the name of the section.
But I guess with the change described below in my current comment, it would totally make sense.

> What is the usecase for end?

Ctrl + A selects everything indeed
But I also want to allow to select one of the date in the date range field to reset it.
So I'm building the blocks to allow selecting several section but not all.

In the updater, we have the following logic:

```tsx
const nexSelectedSectionIndexes = start == null ? null : { start, end: end ?? start };
```

So internal methods just past `start` most of the time.
We could store this "raw" value in the state and instead do the same logic in a `useMemo`.
That way people would only have to care about `end` when it's useful, like inside the hooks right now.

Reply:
I pushed a new version with this approach